src/transformers/trainer.py

```diff
@@ -3875,6 +3912,23 @@ def predict(
             - metrics (`Dict[str, float]`, *optional*): The potential dictionary of metrics (if the dataset contained
               labels).
         """
+        # Set generation-related kwargs
+        if self.args.predict_with_generate:
+            if self.args.generation_config is not None:
+                gen_config = self.args.generation_config
+                self.gen_config = copy.deepcopy(gen_config)  # copy so we don't modify args.gen_config in-place
+                unused_kwargs = self.gen_config.update(**gen_kwargs)
+                if unused_kwargs:
+                    logger.warning_once(
+                        f"Following generation related kwargs were passed to `evaluate` but not used by `generate()`: "
+                        f"{' '.join(unused_kwargs.keys())} .",
+                        "Make sure there are no typos in the passed kwargs or do not pass unused kwargs.",
+                    )
+            else:
+                # We assume the model can generate if predict-with-generate is True
+                # Therefore, generation_config should be available
+                self.gen_config = self.model.generation_config
```

Comment:
shouldn't we also update the config with gen_kwargs ?

Reply:
you mean add kwargs from `model.config` to generation config? It shouldn't be necessary because the base `model.generation_config` should contain all generation related kwargs after the model is loaded. So we just need to make sure user-passed kwargs have higher priority than `trainer.generation_config`

Reply:
I'm taking about the `gen_kwargs` that you are passing in `predict`. I would expect that `self.gen_config` is updated when the user pass `gen_kwargs` in the `predict` function in all cases (important in the case we pass a generate kwargs such as `synced_gpus` ). By default, it is equal to `self.model.generation_config` but if the user passes it in TrainingArguments, it will be equal to `self.args.generation_config`.

Reply:
Ah I see now, right, we should be updating i any case